core/shared/src/main/scala/cats/parse/Parser.scala

```diff
@@ -198,6 +198,16 @@ sealed abstract class Parser[+A] {
   def flatMap[B](fn: A => Parser[B]): Parser[B] =
     Parser.flatMap(this)(fn)
 
+  /** Replaces parsed values with the given value.
+    */
+  def as[B](b: B): Parser[B] =
+    Parser.as(this)(b)
```

Comment:
I think here we can just do `.void.map(_ => b)` since this is not fundamental. And remove from the companion.